File: truevfs-ext/truevfs-ext-logging/src/main/scala/net/java/truevfs/ext/logging/LogInputChannel.scala
/*
 * Copyright (C) 2005-2015 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.logging

import java.nio.channels._
import net.java.truecommons.io._
import net.java.truecommons.cio._
import org.slf4j._

/**
 * @author Christian Schlichtherle
 */
private final class LogInputChannel(
  override val context: InputSocket[_ <: Entry],
  channel: SeekableByteChannel
) extends ReadOnlyChannel(channel) with LogCloseable {

  override def logger: Logger = LogInputChannel.logger
}

private object LogInputChannel {

  val logger: Logger = LoggerFactory.getLogger(classOf[LogInputChannel])
}

File: truevfs-ext/truevfs-ext-logging/src/main/java/net/java/truevfs/ext/logging/LogInputChannel.java
package net.java.truevfs.ext.logging;

import net.java.truecommons.cio.Entry;
import net.java.truecommons.cio.IoSocket;
import net.java.truecommons.io.ReadOnlyChannel;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.nio.channels.SeekableByteChannel;

final class LogInputChannel extends ReadOnlyChannel implements LogCloseable {

    private static final Logger logger = LoggerFactory.getLogger(LogInputChannel.class);

    private final IoSocket<? extends Entry> context;

    LogInputChannel(final IoSocket<? extends Entry> context, final SeekableByteChannel channel) {
        super(channel);
        this.context = context;
        opening();
    }

    @Override
    public IoSocket<? extends Entry> context() {
        return context;
    }

    @Override
    public Logger logger() {
        return logger;
    }
}
